src/jlapi.c

```diff
@@ -318,6 +318,26 @@ JL_DLLEXPORT jl_value_t *(jl_typeof)(jl_value_t *v)
     return jl_typeof(v);
 }
 
+JL_DLLEXPORT int8_t (jl_gc_unsafe_enter)(void)
+{
+    return jl_gc_unsafe_enter();
+}
+
+JL_DLLEXPORT void (jl_gc_unsafe_leave)(int8_t state)
+{
+    jl_gc_unsafe_leave(state);
+}
+
+JL_DLLEXPORT int8_t (jl_gc_safe_enter)(void)
```

Comment:
Well, you are either safe and unsafe but you might not know which one your are in when entering the function. This is important so that when we add codegen support, we can move a GC transition across a runtime function call and merge it with another transition.

The `safe` / `unsafe` `enter` function set the state to safe or unsafe and return the current state, the `safe` / `unsafe` `leave` function then set it back.